packages/x-components/src/components/base-switch.vue

```diff
@@ -0,0 +1,161 @@
+<template>
+  <button
+    @click="toggle"
+    :aria-checked="value.toString()"
+    :class="cssClasses"
+    class="x-switch"
+    role="switch"
+  >
+    <div class="x-switch__handle" />
+  </button>
+</template>
+
+<script lang="ts">
+  import Vue from 'vue';
+  import { Component, Prop } from 'vue-property-decorator';
+  import { VueCSSClasses } from '../utils';
+
+  /**
+   * Basic switch component to handle boolean values. This component receives
+   * its selected state using a prop, and emits an event whenever the user
```

Comment:
Maybe we should specify that the emitted event is an internal (`Vue`) event and not an `XEvent` (which is later specify in the `<docs>` section).